tfhe/src/high_level_api/glwe_ct_list.rs

```diff
@@ -0,0 +1,211 @@
+use super::global_state::with_cpu_internal_keys;
+use crate::core_crypto::commons::math::random::{Deserialize, Serialize};
+use crate::high_level_api::integers::{FheIntId, FheUintId};
+use crate::integer::ciphertext::{DataKind, Expandable, GlwePackable};
+use crate::named::Named;
+use crate::shortint::Ciphertext;
+use crate::{FheBool, FheInt, FheUint};
+
+impl<Id: FheUintId> GlwePackable for FheUint<Id> {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        self.ciphertext.on_cpu().to_owned().compact_into(messages)
+    }
+}
+
+impl<Id: FheIntId> GlwePackable for FheInt<Id> {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        self.ciphertext.on_cpu().to_owned().compact_into(messages)
+    }
+}
+
+impl GlwePackable for FheBool {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        self.ciphertext.on_cpu().to_owned().compact_into(messages)
+    }
+}
+
+pub struct GlwePackedCompressedCiphertextListBuilder {
+    inner: crate::integer::ciphertext::GlwePackedCompressedCiphertextListBuilder,
+}
+
+impl GlwePackedCompressedCiphertextListBuilder {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            inner: crate::integer::ciphertext::GlwePackedCompressedCiphertextListBuilder::new(),
+        }
+    }
+
+    pub fn push<T>(&mut self, value: T) -> &mut Self
+    where
+        T: GlwePackable,
+    {
+        self.inner.push(value);
+        self
+    }
+
+    pub fn extend<T>(&mut self, values: impl Iterator<Item = T>) -> &mut Self
+    where
+        T: GlwePackable,
+    {
+        self.inner.extend(values);
+        self
+    }
+
+    pub fn build(&self) -> GlwePackedCompressedCiphertextList {
+        with_cpu_internal_keys(|key| {
+            GlwePackedCompressedCiphertextList(
+                self.inner.build(
+                    key.compression_key
+                        .as_ref()
+                        .expect("Compression key not set in server key"),
+                ),
+            )
+        })
```

Comment:
No that important but maybe here we could avoid panicking on error cases and return a `crate::Result``

That would mean something like
```
crate::high_level_api::global_state::try_with_internal_keys(|keys| {
   match keys {
       Some(InternalServerKey::Cpu(cpu_key)) => {
           cpu_key.compression_key.as_ref().ok_or_else(|| crate::Error::new("Compression key not set in server key))
       },
      _ => Err(crate::Error::new("A Cpu server key is needed to be set".to_string()))
   }
})
```

Reply:
Done